packages/core/src/routes/interaction/types/hook.ts

```diff
@@ -0,0 +1,19 @@
+import { type InteractionEvent } from '@logto/schemas';
+
+/**
+ * This type is used to hold the hook-related interaction context when processing an interaction that may need to trigger hooks.
+ * In the `koaInteractionHooks` middleware, we will store the context before processing the interaction and consume it after the interaction is processed if needed.
+ */
+export type InteractionHookContext = {
+  event: InteractionEvent;
+  sessionId?: string;
+  applicationId?: string;
+};
+
+/**
+ * This type is used to hold the hook-related interaction results.
```

Comment:
No need for "This type is used to", as the comment clearly points to the exact type